### ISSUE TITLE
The value should not be considered when cleared

### ISSUE THREAD

**Issue by sguptatw** [getgauge/taiko#199]
**Expected behavior**
The value should not be considered when cleared

**Actual behavior**
The previous value is being considered.

**Steps to replicate**
```
const { browser, openBrowser, goto, link, click, listItem,checkBox } = require('taiko');

(async () => {
    try {
        await openBrowser({headless:false});
        await goto("http://todomvc.com/examples/react/#/");
        await write("something");
        await clear()
        await press("Enter");  
        await waitFor(5000)
        await closeBrowser();
    } catch (e) {
        console.error(e);
    } finally {
    }
})();
```
> Value `something` is being considered and the task gets added.

**Version**
```
3306dde13abf2dfc8c83e796b8d33ae0c42b1e01
```
Blocks #65

**Comment by NivedhaSenthil** [comment]
It is very specific to the way todo app handling it. Taiko does not have control over it. They have listeners on `keyDown` and `keyUp` which is causing this effect.

**Comment by nehashri** [comment]
Taiko is trying to clear the text using `document.activeElement.value = ""` [here](https://github.com/getgauge/taiko/blob/master/lib/taiko.js#L631).
This may not be the most ideal way to handle this. 
The write api uses a dispatch event to write the text. The clear api can either reuse that api and press backspace to clear the text or select the entire text in the box and press delete or backspace. This is more aligned with how the users may implement the clear action.

**Comment by NivedhaSenthil** [comment]
If it has to done that way user can always do it with combination of existing APIs, reason why value of an element is set to "" is because the other ways mentioned above are extremely slow as it involves many steps to be done.

**Comment by sguptatw** [comment]
IMO - The option suggested by @nehashri makes sense. It mimics how the user performs it on the UI. Furthermore, this is a convenience API so instead of using combination of existing APIs, users have a simple single API that can be invoked.
Thoughts?

**Comment by NivedhaSenthil** [comment]
In the other issue referring the same https://github.com/getgauge/taiko/issues/191 tells about the delay using the combination. Had to do it the different way because of that, but really have to check the impact both cases.

**Comment by NivedhaSenthil** [comment]
Fix should be available in 252db32. Clear is now simulated like user by selecting using clicking thrice and pressing backspace. Performance seems to be fine as we dont wait for navigation implicitly unlike user doing it by combination of API.
